### ISSUE TITLE
Decimal Numbers Without Leading Zero

### ISSUE THREAD

**Issue by agentile1990** [Berkmann18/json-fixer#58]
## Expected behaviour
Decimal numbers without a leading 0 are throwing an error. 
Example: 
`{"weight":.4}` => `{ weight: 0.4 }`

## Actual behaviour
`Error: Unsupported issue: Expected "," or "}" but end of input found. (please open an issue at the repo)`

## How to reproduce
Run input string through json-fixer. See minimum example repo below. 

## Possible solution(s)
In my specific case, my temporary workaround is 
``` javascript 
str.replace(/\.(?!":)(?=[\d]{1,}\,)/g, ' 0.');
```
Still looking through codebase to determine a fix within the package. 

## Environment

- Version:
- OS(s): Mac OS X 10.15 Catalina Version 10.15.7
- Browser(s) (if applicable): N/A
- Node version: 12.19.0
- NPM version: 6.14.8

## Link or repo

https://github.com/agentile1990/json-fixer-issue

**Comment by Berkmann18** [comment]
H
Thanks for reporting this issue, I've taken note of it and will look into it when I can.
PRs are welcome!

**Comment by agentile1990** [comment]
@Berkmann18 thanks for the quick reply. I'm looking at this as well and would like to contribute. 

Working through the codebase and the sandbox this evening, I discovered that it only happens when the space trailing space after the number is missing: 
`.4 }` works while `.4}` throws the error.

It looks like the first pass misses where it should put the quotes: 
```
Trailing character
Adding quotes...
Nearly fixed data:
0 {"weight": "0.4}"
```
The next time pass it finds the end of input, so it bails. 
`peg$SyntaxError: Expected "," or "}" but end of input found.`

**Comment by agentile1990** [comment]
I was able to track down what was going on today. 

Line 32 in fixer.js
``` javascript
const unquotedWord = /(":\s*)(\S*)/g.exec(fixedLine);
```

When the value is a Number followed by a comma `0.4,` the comma is a part of the match. This causes it to be treated as `NaN` and sent to `quotify()` 

I played around with the regex using positive lookahead, but wasn't able to keep binary / octet fixes happy.

**Comment by Berkmann18** [comment]
It indeed seems that the lack of space between `}` and `0.4` leads to an unfixed JSON code given the experiment I did.
If you've got a (partial) solution, feel free to submit a (draft) PR for it. I may not be able to fully look at myself given some nearby deadlines but I'll make sure I can find some time to look into this.

**Comment by Berkmann18** [comment]
@agentile1990 Did you found a solution, I didn't get a chance to look into it deeper.

**Comment by agentile1990** [comment]
Hi @Berkmann18, 
Sorry, never got a chance to dig into this further. 

We ended up using a different solution all together. We figured out that the invalid JSON we were getting from the application _was valid JavaScript_. We ended up wrapping `eval` in a function something like this: 

``` javascript

function (invalidJson) {
    return eval('(function() {return ' + invalidJson.replace(/[\n\r]+/g, ' ') + '})()');
}
```

**IMPORTANT**: Anyone wanting to use this solution should be aware of the potential security vulnerabilities with this approach. This solution was implemented in a very isolated NodeJS environment.
